docs/src/main/asciidoc/spring-cloud-gateway.adoc

```diff
@@ -1283,72 +1404,71 @@ spring:
+The `KeyResolver` interface lets pluggable strategies derive the key for limiting requests.
+In future milestone releases, there will be some `KeyResolver` implementations.
 
-=== `RewriteLocationResponseHeader` `GatewayFilter` Factory
+The default implementation of `KeyResolver` is the `PrincipalNameKeyResolver`, which retrieves the `Principal` from the `ServerWebExchange` and calls `Principal.getName()`.
 
-The `RewriteLocationResponseHeader` `GatewayFilter` factory modifies the value of the `Location` response header, usually to get rid of backend-specific details.
-It takes `stripVersionMode`, `locationHeaderName`, `hostValue`, and `protocolsRegex` parameters.
-The following listing configures a `RewriteLocationResponseHeader` `GatewayFilter`:
+By default, if the `KeyResolver` does not find a key, requests are denied.
+You can adjust this behavior by setting the `spring.cloud.gateway.filter.request-rate-limiter.deny-empty-key` (`true` or `false`) and `spring.cloud.gateway.filter.request-rate-limiter.empty-key-status-code` properties.
 
-.application.yml
+[NOTE]
+=====
+The `RequestRateLimiter` is not configurable with the "shortcut" notation. The following example below is _invalid_:
+
+.application.properties
 ====
-[source,yaml]
 ----
-spring:
-  cloud:
-    gateway:
-      routes:
-      - id: rewritelocationresponseheader_route
-        uri: http://example.org
-        filters:
-        - RewriteLocationResponseHeader=AS_IN_REQUEST, Location, ,
+# INVALID SHORTCUT CONFIGURATION
+spring.cloud.gateway.routes[0].filters[0]=RequestRateLimiter=2, 2, #{@userkeyresolver}
 ----
 ====
+=====
 
-For example, for a request of `POST https://api.example.com/some/object/name`, the `Location` response header value of `https://object-service.prod.example.net/v2/some/object/id` is rewritten as `https://api.example.com/some/object/id`.
+==== The Redis `RateLimiter`
 
-The `stripVersionMode` parameter has the following possible values: `NEVER_STRIP`, `AS_IN_REQUEST` (default), and `ALWAYS_STRIP`.
+The Redis implementation is based off of work done at https://stripe.com/blog/rate-limiters[Stripe].
+It requires the use of the `spring-boot-starter-data-redis-reactive` Spring Boot starter.
 
- * `NEVER_STRIP`: The version is not stripped, even if the original request path contains no version.
- * `AS_IN_REQUEST` The version is stripped only if the original request path contains no version.
- * `ALWAYS_STRIP` The version is always stripped, even if the original request path contains version.
+The algorithm used is the https://en.wikipedia.org/wiki/Token_bucket[Token Bucket Algorithm].
 
-The `hostValue` parameter, if provided, is used to replace the `host:port` portion of the response `Location` header.
-If it is not provided, the value of the `Host` request header is used.
+The `redis-rate-limiter.replenishRate` property is how many requests per second you want a user to be allowed to do, without any dropped requests.
```

Comment:
Change to:

The `redis-rate-limiter.replenishRate` property defines how many requests per second to allow (without any dropped requests).